components/gsma-authenticators/src/main/java/com/wso2telco/gsma/authenticators/voice/VoiceCallAuthenticator.java

```diff
@@ -0,0 +1,265 @@
+
+import com.wso2telco.Util;
+import com.wso2telco.gsma.authenticators.AuthenticatorException;
+import com.wso2telco.gsma.authenticators.Constants;
+import com.wso2telco.gsma.authenticators.DBUtils;
+import com.wso2telco.gsma.authenticators.util.AdminServiceUtil;
+import com.wso2telco.gsma.authenticators.util.AuthenticationContextHelper;
+import com.wso2telco.gsma.authenticators.util.BasicFutureCallback;
+import com.wso2telco.gsma.authenticators.util.UserProfileManager;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.client.HttpClient;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.util.EntityUtils;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.wso2.carbon.identity.application.authentication.framework.AbstractApplicationAuthenticator;
+import org.wso2.carbon.identity.application.authentication.framework.AuthenticatorFlowStatus;
+import org.wso2.carbon.identity.application.authentication.framework.LocalApplicationAuthenticator;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.ApplicationConfig;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.exception.LogoutFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.user.registration.stub.UserRegistrationAdminServiceIdentityException;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.rmi.RemoteException;
+import java.sql.SQLException;
+import java.util.Map;
+
+/**
+ * Created by sheshan on 4/26/17.
+ */
+public class VoiceCallAuthenticator extends AbstractApplicationAuthenticator
+        implements LocalApplicationAuthenticator {
+
+    private static final String MSISDN = "msisdn";
+    private static final String CLIENT_ID = "relyingParty";
+    private static final String IS_FLOW_COMPLETED = "isFlowCompleted";
+    private static Log log = LogFactory.getLog(VoiceCallAuthenticator.class);
+
+    private final String isUserEnrolledUrl = "https://poc.vsservic.es/pwr/isUserActive";
+    private final String verifyUserUrl = "https://poc.vsservic.es/pwr/authenticateUser";
+    private final String onBoardUserUrl = "https://poc.vsservic.es/pwr/onboardUser";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this final field static too. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.wso2telco.com:9000/coding_rules#rule_key=squid%3AS1170)